src/main/java/edu/usc/ir/geo/gazetteer/GeoNameResolver.java

```diff
@@ -604,7 +604,12 @@ public static void main(String[] args) throws Exception {
 				//TODO: get port from CLI args
 				int port = 8765;
 				Launcher.launchService(port, indexPath);
-			} else {
+			}else if (!line.hasOption("server") &&
+				!line.hasOption("search") &&
+				!line.hasOption("build") &&
+				!line.hasOption("index") &&
+				!line.hasOption("help")) 
+			{
 				System.err.println("Sub command not recognised");
 				System.exit(-1);
```

Comment:
That resolves a old bug @MBoustani :)

@chrismattmann How about removing all "System.exit" ? Shouldn't we just use "return" for success and RuntimeException for error. "System.exit" terminates the JVM abnormally and I think can be replaced